src/main.py

```diff
@@ -33,27 +39,45 @@
 )
 
 current_wattages = []
+last_current_data_timestamp = datetime.now()
+last_total_data_timestamp = datetime.now()
 
 def process_reading(watt_current: float, watt_total_low: float, watt_total_high: float, total_gas: float) -> None:
-  global current_wattages
-  current_date = datetime.now().isoformat()
+  global current_wattages, last_current_data_timestamp, last_total_data_timestamp
+  global current_data_interval_seconds, total_data_interval_seconds
+  current_date = datetime.now()
   current_wattages.append(watt_current)
-  if len(current_wattages) == 10:
-    # done with a minute, lets write
+
+  if current_data_interval_seconds is None or last_current_data_timestamp + current_data_interval_seconds < current_date:
+    last_current_data_timestamp = current_date
     avg = mean(current_wattages)
     json_body = [
       {
           "measurement": "energy",
-          "time": current_date,
+          "time": current_date.isoformat(),
+          "fields": {
+              "current_electricity_kw": float(avg)
```

Comment:
this isnt going to scale with gas added, and its called `energy` as a measurement, is that allowed?

(this might be my understanding of influx) but isn't the measurement a "table", in which you need to have a certain format?

Reply:
We can split it up, but Influx allows this. Basically, each of the fields is a kind of column in the table that is this measurement. On some rows, the other columns are just empty.